scripts/run_code_generation.py

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python3
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+import argparse
+import concurrent
+import concurrent.futures
+import datetime
+import io
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import zipfile
+from concurrent.futures import wait, FIRST_COMPLETED, ALL_COMPLETED
+from pathlib import Path
+
+CLIENT_MODEL_FILE_LOCATION = "./code-generation/api-descriptions/"
+DEFAULTS_FILE_LOCATION = "../defaults/sdk-default-configuration.json"  # Relative to models dir
+DEFAULT_GENERATOR_LOCATION = "code-generation/generator/"
+GENERATOR_JAR = "target/aws-client-generator-1.0-SNAPSHOT-jar-with-dependencies.jar"
+
+
+SERVICE_MODEL_FILENAME_PATTERN = re.compile(
+    "^"
+    "(?P<service>.+)-"                                      # service name
+    "(?P<date>[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9])"  # model date
+    ".normal.json$"
+)
+
+SERVICE_NAME_REMAPS = {"runtime.lex": "lex",
+                       "runtime.lex.v2": "lexv2-runtime",
+                       "models.lex.v2": "lexv2-models",
+                       "transfer": "awstransfer",
+                       "transcribe-streaming": "transcribestreaming",
+                       "streams.dynamodb": "dynamodbstreams"}
+
+
+def collect_available_models(models_dir: str):
+    model_files = os.listdir(models_dir)
+    service_name_to_model_filename_date = dict()
+
+    for filename in model_files:
+        if not os.path.isfile("/".join([models_dir, filename])):
+            continue
+        match = SERVICE_MODEL_FILENAME_PATTERN.match(filename)
+
+        service_model_name = match.group("service")
+        service_model_date = match.group("date")
+        service_model_date = datetime.datetime.strptime(service_model_date, "%Y-%m-%d").date()
+        already_found_model = service_name_to_model_filename_date.get(service_model_name, None)
+        if already_found_model:
+            already_found_date = already_found_model[1]
+            if already_found_date < service_model_date:
+                service_name_to_model_filename_date[service_model_name] = (filename, service_model_date)
+        else:
+            service_name_to_model_filename_date[service_model_name] = (filename, service_model_date)
+
+    service_name_to_model_filename = dict()
+    for raw_key, model_file_date in service_name_to_model_filename_date.items():
+        key = SERVICE_NAME_REMAPS.get(raw_key, raw_key)
+        if "." in key:
+            key = "-".join(reversed(key.split(".")))  # just replicating existing legacy behavior
+        if ";" in key:
+            key = key.replace(";", "-")  # just in case... just replicating existing legacy behavior
+
+        service_name_to_model_filename[key] = model_file_date[0]
+        if key == "s3":
+            service_name_to_model_filename["s3-crt"] = model_file_date[0]
+
+    # for key, val in service_name_to_model_filename.items():
```

Comment:
nit pick: we could remove the commented out code

Reply:
done